app/src/main/java/com/karumi/ui/presenter/SuperHeroDetailPresenter.kt

```diff
@@ -25,20 +28,25 @@ class SuperHeroDetailPresenter(
 
     override fun update() {
         view?.showLoading()
-        refreshSuperHeroes()
+        refreshSuperHero()
     }
 
-    private fun refreshSuperHeroes() = async {
+    private fun refreshSuperHero() = async {
         val result = await { getSuperHeroByName(name) }
         view?.hideLoading()
-        view?.showSuperHero(result)
+        when (result) {
+            is Right -> view?.showSuperHero(result.r)
+            is Left -> view?.showError(result.l)
+        }
+
     }
 
     interface View {
         fun close()
         fun showLoading()
         fun hideLoading()
         fun showSuperHero(superHero: SuperHero)
+        fun showError(l: DomainError)
```

Comment:
Also, do we need it to be a `DomainError`? Maybe it'd be better if it'd be a plain string instead?

Reply:
this is one of my eternal questions it's responsability of the presenter map to the string or R.string or it's presponsability of the view, if a return a string i need a referece to the context here.